irepela/homework_9/fraction.py

```diff
@@ -0,0 +1,97 @@
+from math import gcd
+from functools import total_ordering
+
+
+@total_ordering
+class Fraction:
+
+    def __init__(self, num: int, denom: int = 1):
+
+        if denom == 0:
+            raise ZeroDivisionError
+
+        _gcd = gcd(num, denom)
+        self.num = num // _gcd
+        self.denom = denom // _gcd
+
+        if self.denom < 0:
+            self.num *= -1
+            self.denom *= -1
+
+    def __repr__(self):
+        return f"Fraction({self.num}, {self.denom})"
+
+    def __str__(self):
+        return f"'{self.num} / {self.denom}'"
+
+    def __format__(self, fmt):
+        if fmt == "dec":
+            return str(self.num / self.denom)
+        return str(self)
+
+    def __bool__(self):
+        return bool(self.num)
+
+    def __eq__(self, other):
+        if isinstance(other, Fraction):
+            return self.num == other.num and self.denom == other.denom
+        elif isinstance(other, int):
+            return self == Fraction(other)
+        else:
+            raise NotImplementedError
```

Comment:
try to decouple parts of function:

- check if element can be converted to fraction
- convert element to fraction
- perform action between 2 fractions

this approach can unify process and help extend Fraction in future e.g. in case if you want to support str or float

Reply:
Also I would suggest to avoid many returns in if .. else block

Reply:
it makes sense to create separate classmethod:
 can_convert(item) -> bool